# Práctica 1.1 — Probabilidad y Estadística
## Análisis de la evolución poblacional mundial

## Objetivo

Utilizar un conjunto de datos real para comprobar el funcionamiento del entorno de análisis de datos y aplicar medidas estadísticas descriptivas básicas utilizando Pandas y NumPy.

## Actividad 1 — Identificación de la fuente

**Propósito:** Reconocer la procedencia del conjunto de datos antes de utilizarlo y documentar correctamente su fuente.

Los datos utilizados en esta práctica provienen de **Our World in Data (OWID)**.

La fuente original es **United Nations, World Population Prospects (2024)**.

**Cita:**

> UN, World Population Prospects (2024) – processed by Our World in Data. "Annual change in population – UN WPP" [dataset]. United Nations, "World Population Prospects"; United Nations, "World Population Prospects - Interim Update" [original data].

### Preguntas

**¿Quién produjo originalmente los datos?**
Las Naciones Unidas (UN), a través de su informe *World Population Prospects (2024)*.

**¿Qué organización los procesa para su utilización en Our World in Data?**
Our World in Data (OWID) 

**¿Qué representa la variable que analizarás?**
La variable `Annual change in population` representa el cambio neto anual de población 

## Actividad 2 — Carga y exploración de datos

**Propósito:** Comprobar que el entorno de Python, Pandas y Jupyter funciona correctamente y realizar una primera exploración del dataset.

### 2.1 Importar las bibliotecas

In [1]:
import pandas as pd
import numpy as np

### 2.2 Cargar el archivo CSV

In [2]:
df = pd.read_csv("../data/raw/annual-population-growth/annual-population-growth.csv")
df.head()

,Entity,Code,Year,Annual change in population,Annual population change (Projected)
0,Afghanistan,AFG,1951,103163.0,NaN
1,Afghanistan,AFG,1952,108441.0,NaN
2,Afghanistan,AFG,1953,108919.0,NaN
3,Afghanistan,AFG,1954,111251.0,NaN
4,Afghanistan,AFG,1955,119027.0,NaN


### 2.3 Explorar el DataFrame

In [3]:
# Primeras filas
df.head()

,Entity,Code,Year,Annual change in population,Annual population change (Projected)
0,Afghanistan,AFG,1951,103163.0,NaN
1,Afghanistan,AFG,1952,108441.0,NaN
2,Afghanistan,AFG,1953,108919.0,NaN
3,Afghanistan,AFG,1954,111251.0,NaN
4,Afghanistan,AFG,1955,119027.0,NaN


In [4]:
# Dimensiones
df.shape

(38400, 5)

In [5]:
# Nombres de columnas
df.columns.tolist()

['Entity',
 'Code',
 'Year',
 'Annual change in population',
 'Annual population change (Projected)']

In [6]:
# Tipos de datos
df.dtypes

Entity                                      str
Code                                        str
Year                                      int64
Annual change in population             float64
Annual population change (Projected)    float64
dtype: object

In [7]:
# Estadísticas descriptivas
df.describe()

,Year,Annual change in population,Annual population change (Projected)
count,38400.000000,1.868800e+04,1.971200e+04
mean,2025.500000,2.063019e+06,8.375481e+05
std,43.300872,9.440859e+06,5.816745e+06
min,1951.000000,-3.315935e+06,-2.291035e+07
25%,1988.000000,1.781000e+03,-9.742750e+03
50%,2025.500000,4.689800e+04,9.350000e+01
75%,2063.000000,3.021128e+05,1.411160e+05
max,2100.000000,9.337137e+07,7.206736e+07


In [8]:
# Cantidad de valores faltantes
df.isnull().sum()

Entity                                      0
Code                                     1200
Year                                        0
Annual change in population             19712
Annual population change (Projected)    18688
dtype: int64

In [9]:
# Cantidad de entidades
df["Entity"].nunique()

256

### Preguntas

**¿Cuántas filas y columnas tiene el dataset?**
El dataset tiene 38,400 filas y 5 columnas.

**¿Qué tipo de datos contiene la variable `Year`?**
Es un dato de tipo Int  que representa el año como valor

**¿Existen valores faltantes?**
Sí. La columna `Code` tiene 1,200 valores faltantes, `Annual change in population` tiene 19,712 valores faltantes y `Annual population change (Projected)` tiene 18,688

**¿Cuántas entidades diferentes contiene el dataset?**
El dataset contiene 256 entidades diferentes

## Actividad 3 — Estadística descriptiva

**Propósito:** Aplicar medidas estadísticas básicas a una variable poblacional.

Se selecciona el año **2023** (último año con datos históricos, ya que el rango histórico abarca de 1951 a 2023) y se filtran los registros correspondientes sobre la variable `Annual change in population`.

In [10]:
col = "Annual change in population"

# Se filtran solo los datos históricos (no nulos) para el año seleccionado
df_2023 = df[(df["Year"] == 2023) & (df[col].notnull())]
df_2023.head()

,Entity,Code,Year,Annual change in population,Annual population change (Projected)
72,Afghanistan,AFG,2023,875915.0,NaN
222,Africa (UN),UN_AFR,2023,33886868.0,NaN
372,Albania,ALB,2023,-15954.0,NaN
522,Algeria,DZA,2023,686831.0,NaN
672,American Samoa,ASM,2023,-821.0,NaN


In [11]:
media = df_2023[col].mean()
mediana = df_2023[col].median()
varianza = df_2023[col].var()
desviacion = df_2023[col].std()
minimo = df_2023[col].min()
maximo = df_2023[col].max()

print("Media:", media)
print("Mediana:", mediana)
print("Varianza:", varianza)
print("Desviación estándar:", desviacion)
print("Mínimo:", minimo)
print("Máximo:", maximo)

Media: 2008097.765625
Mediana: 46865.5
Varianza: 82427560967665.19
Desviación estándar: 9078962.549083743
Mínimo: -3315935.0
Máximo: 72315107.0


In [12]:
entidad_min = df_2023.loc[df_2023[col].idxmin(), "Entity"]
entidad_max = df_2023.loc[df_2023[col].idxmax(), "Entity"]

print("Entidad con el valor mínimo:", entidad_min)
print("Entidad con el valor máximo:", entidad_max)

Entidad con el valor mínimo: Ukraine
Entidad con el valor máximo: Less developed regions, excluding China


### Preguntas

**¿Cuál es el valor de la media?**
La media del cambio anual de población en 2023 es de aproximadamente 2,008,097.77 personas.

**¿Cuál es el valor de la mediana?**
La mediana es de aproximadamente 46,865.5 personas

**¿Cuál es el valor de la desviación estándar?**
La desviación estándar es de aproximadamente **9,078,962.55** personas, 
**¿Qué entidad presenta el valor mínimo y cuál el máximo?**
El valor mínimo corresponde a **Ucrania**  y el valor máximo corresponde a **"Less developed regions, excluding China"**

## Actividad 4 — Comprobación del entorno

**Propósito:** Comprobar que el notebook está utilizando correctamente las herramientas configuradas para la práctica.

In [13]:
import sys

print("Python:", sys.version)
print("Pandas:", pd.__version__)
print("NumPy:", np.__version__)

Python: 3.14.7 (tags/v3.14.7:823f032, Aug  5 2026, 10:51:32) [MSC v.1944 64 bit (AMD64)]
Pandas: 3.0.5
NumPy: 2.5.2


## Resultados

Este notebook mostró:

- Información básica de la fuente de los datos (UN World Population Prospects, procesados por OWID).
- La cita completa de la fuente.
- El DataFrame cargado correctamente desde `annual-population-growth.csv`.
- Información sobre su estructura (dimensiones, columnas, tipos de datos, valores faltantes, entidades).
- Resultados de los estadísticos descriptivos (media, mediana, varianza, desviación estándar, mínimo y máximo) para el año 2023.
- Las versiones de Python, Pandas y NumPy utilizadas en el entorno.